jupyterhub_traefik_proxy/etcd.py

```diff
@@ -57,24 +64,30 @@ def _default_client(self):
         help="""the etcd key prefix for traefik configuration""",
     )
 
-    def _setup_etcd(self):
+    async def _setup_etcd(self):
         self.log.info("Seting up etcd...")
-        self.client.put(self.etcd_traefik_prefix + "debug", "true")
-        self.client.put(self.etcd_traefik_prefix + "defaultentrypoints/0", "http")
-        self.client.put(
+        await self.etcd_client.put(self.etcd_traefik_prefix + "debug", "true")
```

Comment:
Can this setup be a single transaction as well?

Reply:
Yes it should be, sorry, I've missed it.